docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md

```diff
@@ -0,0 +1,251 @@
+- Aiven for Apache Kafka service with Apache Kafka Connect set up and running.
+  For setup instructions, see
+  [Aiven for Apache Kafka with Kafka Connect documentation](https://aiven.io/docs/products/kafka/kafka-connect/get-started).
+- [Aiven CLI](/docs/tools/cli).
+- [HashiCorp Vault address and token](https://developer.hashicorp.com/vault/docs/concepts/tokens).
+
+## Configure secret providers
+
+Set up HashiCorp Vault in your Aiven for Apache Kafka Connect service to manage and
+access sensitive information.
+
+<Tabs groupId="config-methods">
+<TabItem value="api" label="API" default>
+
+Use the [ServiceUpdate](https://api.aiven.io/doc/#tag/Service/operation/ServiceUpdate)
+API to update your service configuration. Add the HashiCorp Vault configuration to
+the `user_config` using the following API request:
+
+```sh
+curl --request PUT \
+  --url https://api.aiven.io/v1/project/{PROJECT_NAME}/service/{SERVICE_NAME} \
+  --header 'Authorization: Bearer YOUR_BEARER_TOKEN' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "user_config": {
+      "secret_providers": [
+        {
+          "name": "vault",
+          "vault": {
+            "auth_method": "token",
+            "address": "https://vault.aiven.fi:8200/
+            "token": "YOUR_VAULT_TOKEN"
+          }
+        }
+      ]
+    }
+  }'
+```
+
+Parameters:
+
+- `PROJECT_NAME`: Name of your Aiven project.
+- `SERVICE_NAME`: Name of your Aiven Kafka service.
+- `url`: API endpoint for updating service configuration. Replace `{project_name}` and
+  `{service_name}` with your actual project and service names.
+- `Authorization`: Header for authentication. Replace `YOUR_BEARER_TOKEN` with your
+  actual  [Aiven API token](/docs/platform/howto/create_authentication_token).
+- `Content-Type`: Specifies that the request body is in JSON format.
+- `auth_method`: Authentication method used by HashiCorp Vault. In this case, it is token.
+- `address`: Address of the HashiCorp Vault server.
+- `token`: Your HashiCorp Vault token.
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+Configure HashiCorp Vault as a secret provider using [Aiven CLI](/docs/tools/cli):
+
+```sh
+avn service update SERVICE_NAME \
+  -c secret_providers='[
+    {
+        "vault": {
+            "auth_method": "token",
+            "address": "https://vault.aiven.fi:8200/",
+            "token": "YOUR_VAULT_TOKEN"
+        },
+        "name": "vault"
+    }
+  ]'
+```
+
+Parameters:
+
+- `SERVICE_NAME`: Name of your Aiven Kafka service..
+- `name`: Name of the secret provider. In this case, `vault`.
+- `auth_method`: Authentication method used by HashiCorp Vault. In this case, it is
+  `token`.
+- `address`: Address of the HashiCorp Vault server.
+- `token`: Your HashiCorp Vault token.
+
+</TabItem>
+</Tabs>
+
+## Reference secrets in connector configurations
+
+After configuring the secret providers, you can reference secrets in your
+connector configurations.
+
+### JDBC sink connector
```

Comment:
the JDBC example is illustrative right? Any connector we have can support this @keejon ?

Reply:
jup, can be used in any connector 👍

Reply:
Updated the text to clarify this.